Tests/SonarScanner.MSBuild.Test/ArgumentProcessorTests.cs

```diff
@@ -22,364 +22,339 @@
-            logger.AssertWarningsLogged(1);
-            AssertExpectedChildArguments(settings);
-
-            // 5. Partial match -> unrecognized -> treated as preprocessing
-            logger = new TestLogger();
-            settings = CheckProcessingSucceeds(logger, "endx");
-            AssertExpectedPhase(AnalysisPhase.PreProcessing, settings);
-            logger.AssertWarningsLogged(1);
-        }
-
-        [TestMethod]
-        public void ArgProc_EndVerb_MatchesOnlyCompleteWord()
-        {
-            // Arrange
-            TestLogger logger;
-            IBootstrapperSettings settings;
-            logger = new TestLogger();
-
-            // Act
-            // "end", "endx" should not be treated as duplicates
-            settings = CheckProcessingSucceeds(logger, "end", "endX", "endXXX");
-
-            // Assert
-            AssertExpectedPhase(AnalysisPhase.PostProcessing, settings);
-            logger.AssertWarningsLogged(0);
-            AssertExpectedChildArguments(settings, "endX", "endXXX");
-        }
-
-        [TestMethod]
-        public void ArgProc_BeginAndEndVerbs()
-        {
-            // 0. Setup
-            TestLogger logger;
-            var validUrl = "/d:sonar.host.url=http://foo";
-
-            // 1. Both present
-            logger = CheckProcessingFails(validUrl, "begin", "end");
-            logger.AssertErrorsLogged(1);
-            logger.AssertSingleErrorExists("begin", "end");
-        }
-
-        [TestMethod]
-        public void ArgProc_SonarVerbose_IsBool()
-        {
-            var logger = new TestLogger();
+    [TestMethod]
+    public void ArgProc_BeginVerb()
+    {
+        TestLogger logger;
+
+        // 1. Minimal parameters -> valid
+        logger = new();
+        var settings = CheckProcessingSucceeds(logger, ValidUrl, "begin");
+        AssertExpectedPhase(AnalysisPhase.PreProcessing, settings);
+        logger.AssertWarningsLogged(0);
+        AssertExpectedChildArguments(settings, ValidUrl);
+
+        // 2. With additional parameters -> valid
+        logger = new();
+        settings = CheckProcessingSucceeds(logger, ValidUrl, "begin", "ignored", "k=2");
+        AssertExpectedPhase(AnalysisPhase.PreProcessing, settings);
+        logger.AssertWarningsLogged(0);
+        AssertExpectedChildArguments(settings, ValidUrl, "ignored", "k=2");
+
+        // 3. Multiple occurrences -> error
+        logger = CheckProcessingFails(ValidUrl, "begin", "begin");
+        logger.AssertSingleErrorExists(ArgumentProcessor.BeginVerb);
+
+        // 4. Missing -> invalid (missing verb)
+        logger = CheckProcessingFails(ValidUrl);
+        logger.AssertErrorLogged(Resources.ERROR_CmdLine_NeitherBeginNorEndSupplied);
+
+        // 5. Incorrect case -> treated as unrecognized argument -> invalid (missing verb)
+        logger = CheckProcessingFails(ValidUrl);
+        logger.AssertErrorLogged(Resources.ERROR_CmdLine_NeitherBeginNorEndSupplied);
```

Comment:
This assertion appears to be equal to the previous one (lines `203`-`204`). It might be a result of simplifying the old code, maybe `logger = CheckProcessingFails(ValidUrl, "BEGIN");` ?

Reply:
Indeed, `"BEGIN"` was missing. Fixed.